commercetools-models/src/test/java/io/sphere/sdk/carts/CartFixtures.java

```diff
@@ -155,6 +158,59 @@ public static void withLineItemAndCustomLineItemFilledCart(final BlockingSphereC
         });
     }
 
+    public static void withDiscountedLineItem(final BlockingSphereClient client, final RelativeCartDiscountValue relativeCartDiscountValue, final Consumer<Cart> consumer) {
+        withTaxedProduct(client, product -> {
+            withCart(client, (cart) -> {
+                withCustomer(client, (customer) -> {
+                    CartDiscountFixtures.withCartDiscount(client, builder -> builder
+                            .cartPredicate(CartDiscountPredicate.of("customer.id=\"" + customer.getId() + "\""))
+                            .value(relativeCartDiscountValue)
+                            .target(LineItemsTarget.of("product.id=\"" + product.getId() + "\"")), cartDiscount -> {
+                        assertThat(cart.getLineItems()).hasSize(0);
+                        final long quantity = 3;
+                        final String productId = product.getId();
+                        final AddLineItem addLineItemAction = AddLineItem.of(productId, 1, quantity);
+                        final List<UpdateActionImpl<Cart>> updateActions =
+                                asList(addLineItemAction, SetCustomerId.of(customer.getId()), Recalculate.of().withUpdateProductData(true));
+                        final Cart updatedCart = client.executeBlocking(CartUpdateCommand.of(cart, updateActions));
+                        consumer.accept(updatedCart);
+                    });
+                });
+                final Cart cartToDelete = client.executeBlocking(CartByIdGet.of(cart));
```

Comment:
if you use directly `final UnaryOperator<Cart> op` instead of `final Consumer<Cart> consumer` in the parameter list of the function you can save the call to get the current cart version